File: HashMap/HashMap.py
class Node:
    def __init__(self, key, val):
        self.key=key
        self.val=val
        self.next= None
        
class HashMap:
    def __init__(self, capacity=8):
        self.cap= capacity
        self.size=0
        self.buckets= [None]*self.cap
        
    def _hash(self, key):
        return hash(key)%self.cap
    
    def put(self,key,val):
        idx= self._hash(key)
        
        head= self.buckets[idx]
        
        curr=head
        
        while curr:
            if curr.key==key:
                curr.val= val
                return
            curr= curr.next
            
        new_node= Node(key, val)
        
        new_node.next= head
        self.buckets[idx]= new_node
        self.size+=1
        
    def get(self,key, res):
        idx= self.hashKey(key)
        curr= self.buckets[idx]
        
        while curr:
            if curr.key==key:
                res= curr.val
                return True
            curr= curr.next
        return False
    
    def remove(self, key, val):
        idx= self.hashKey(key)
        curr= self.buckets[idx]
        prev= None
        
        while curr:
            if curr.key==key:
                if prev:
                    prev.next=curr.next
                else:
                    self.buckets[idx]= curr.next
                
                self.size-=1
                return
            
            prev= curr
            curr= curr.next
            
    def loadFactor(self):
        return self.size/self.cap

File: HashMap/HashMap.cpp
#include<vector>
#include<iostream>
using namespace std;

template <typename K, typename V>
struct Node
{
    K key;
    V val;

    Node* next= nullptr;

    Node(K k, V v): key(k), val(v), next(nullptr)
};
template <typename K, typename V>
class HashMap
{
    private:
        int cap;
        int size;
        vector<Node<K,V>*> buckets;
        int hashKey(const K& key)
        {
            return hash<K>{}(key)%cap;
        }
    public:
        HashMap(int capacity=8): cap(capacity), size(0)
        {
            buckets.resize(cap, nullptr);
        }

        void put(const K& key, const V& val)
        {
            int idx= hashKey(key);

            Node<K,V>* curr= buckets[idx];

            while(curr)
            {
                if(curr->key==key)
                {
                    curr->val= val;
                    return
                }
                curr= curr->next;
            }
            Node<K,V>* newNode= new Node<K,V>(key,value);
            newNode->next= buckets[idx];
            buckets[idex]= newNode;
            size++;
        }

        bool get(const K& key, V& res)
        {
            int idx= hashKey(key);
            
            Node<K,V>* curr= buckets[idx];

            while(curr)
            {
                if (curr->key==key)
                {
                    res= curr->val;
                    return true;
                }
                curr= curr->next;
            }
            return false;
        }

        void remove(const K& key)
        {
            int idx= hashKey(key);
            Node<V,K>* curr= buckets[idx];
            Node<V,K>* prev= nullptr;

            while (curr)
            {
                if (curr->key==key)
                {
                    if (prev)
                    {
                        prev->next= curr->next;
                    }
                    else buckets[idx]= curr->next;
                    delete curr;
                    size--;
                    return;
                }
                prev= curr;
                curr=curr->next;
            }
        }

        float loadfactor()
        {
            return size/cap;
        }
};